packages/context-editor/src/ContextEditor.tsx

```diff
@@ -98,8 +102,12 @@ export default function ContextEditor(props: ContextEditorProps) {
 						</div>
 					)}
 					<ProseMirrorDoc />
-					<AttributePanel menuHidden={!!props.hideMenu} />
-					<SuggestPanel suggestData={suggestData} setSuggestData={setSuggestData} />
+					<AttributePanel menuHidden={!!props.hideMenu} containerId={containerId} />
```

Comment:
Instead of passing a container ID to the `AttributePanel`, could you instead pass a ref to the container div as a prop? Better yet, what about computing the values you need in the `ContextEditor` (like the `offsetTop`/`clientWidth`) and passing those as scalar props?